src/app-logic/url-handling.js

```diff
@@ -42,27 +42,45 @@ export const CURRENT_URL_VERSION = 5;
  * to manage this bit of state.
  */
 let _isReplaceState: boolean = false;
+let _replaceHistoryCallCount: number = 0;
+
+function _enableHistoryReplaceState(): void {
```

Comment:
Here is an comment I wrote to suggest to export the lower-level functions. But when I looked at the code samples to prove my point, I found that the code sample I thought would be bad was actually better (the second one).

So I'm leaving here just FYI on my train of thoughts and to show this pattern, but I _don't_ suggest to do any change now.


**Here is the full outdated comment:**
Actually I think I'd still export these lower-level functions.

As an example in the not-working patch I was working on, I had this pattern:
```js
if (condition) {
  setHistoryReplaceState(true)
}

// code block 1

setHistoryReplaceState(true)

// code block 2

setHistoryReplaceState(false)
```

This is less straightforward when having only `withHistoryReplaceState`:
```js
if (condition) {
  withHistoryReplaceState(() => {
    // code block 1
  });    
} else {
  // code block 1
}

withHistoryReplaceState(() => {
  // code block 2
});

```

If the new counting functions are exported this could be like this:
```js
if (condition) {
  enableHistoryReplaceState()
}

// code block 1

if (!condition) {
  // The tool is counting each enable so the condition is needed so that it's not called twice.
  enableHistoryReplaceState()
}

// code block 2

maybeDisableHistoryReplaceState()
```